File: src/main/java/io/github/bot/terminal/application/domains/bot_commands/rest/BotCommandsRestConverter.java
package io.github.bot.terminal.application.domains.bot_commands.rest;


import io.github.bot.terminal.application.domains.bot_commands.repository.BotCommandDetails;
import io.github.bot.terminal.application.domains.bot_commands.rest.dto.BotCommandDTO;
import io.github.bot.terminal.application.domains.bot_commands.rest.requests.BotCommandRequest;
import io.github.bot.terminal.application.domains.bot_commands.values.BotCommandState;
import io.github.bot.terminal.application.domains.common.action.ActionRestConverter;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;


@Service
@RequiredArgsConstructor
public class BotCommandsRestConverter {

    private final ActionRestConverter actionRestConverter;

    BotCommandDetails mapToDetails(BotCommandRequest request) {
        BotCommandDetails details = new BotCommandDetails();
        details.setName(request.getName());
        details.setCmd(request.getCmd());
        details.setState(BotCommandState.resolve(request.isEnabled()));
        details.setBotIds(request.getBotIds());
        details.setActionDetails(actionRestConverter.mapToDetails(request.getAction()));
        return details;
    }


    BotCommandDTO mapToDto(BotCommandDetails details) {
        BotCommandDTO dto = new BotCommandDTO();
        dto.setId(details.getId());
        dto.setName(details.getName());
        dto.setCmd(details.getCmd());
        dto.setEnabled(details.getState().isEnabled());
        dto.setBotIds(details.getBotIds());
        dto.setAction(actionRestConverter.mapToDto(details.getActionDetails()));
        return dto;
    }

}

File: src/main/java/io/github/bot/terminal/application/domains/bot_commands/rest/BotCommandsRestConverter.kt
package io.github.bot.terminal.application.domains.bot_commands.rest

import io.github.bot.terminal.application.domains.bot_commands.repository.BotCommandDetails
import io.github.bot.terminal.application.domains.bot_commands.rest.dto.BotCommandDTO
import io.github.bot.terminal.application.domains.bot_commands.rest.requests.BotCommandRequest
import io.github.bot.terminal.application.domains.common.action.ActionRestConverter
import org.springframework.stereotype.Service

@Service
class BotCommandsRestConverter(
        private val actionRestConverter: ActionRestConverter
) {
    fun mapToDetails(request: BotCommandRequest): BotCommandDetails =
            BotCommandDetails(
                    name = request.name,
                    cmd = request.cmd,
                    isEnabled = request.enabled,
                    botIds = request.botIds,
                    actionDetails = actionRestConverter.mapToDetails(request.action)
            )

    fun mapToDto(details: BotCommandDetails): BotCommandDTO =
            BotCommandDTO(
                    id = details.id,
                    name = details.name,
                    cmd = details.cmd,
                    enabled = details.isEnabled,
                    botIds = details.botIds,
                    action = actionRestConverter.mapToDto(details.actionDetails)
            )

}